### DISCUSSION TITLE
v1.0.0-preview2

### DISCUSSION THREAD

**Discussion by mconnew** [CoreWCF/CoreWCF#613] [Releases]
# Release Notes

## Package Info

[CoreWCF.Primitives 1.0.0-preview2](https://www.nuget.org/packages/CoreWCF.Primitives/1.0.0-preview2)  
[CoreWCF.Http 1.0.0-preview2](https://www.nuget.org/packages/CoreWCF.Http/1.0.0-preview2)  
[CoreWCF.NetTcp 1.0.0-preview2](https://www.nuget.org/packages/CoreWCF.NetTcp/1.0.0-preview2)  
[CoreWCF.WebHttp 1.0.0-preview2](https://www.nuget.org/packages/CoreWCF.WebHttp/1.0.0-preview1)  
[CoreWCF.ConfigurationManager 1.0.0-preview2](https://www.nuget.org/packages/CoreWCF.ConfigurationManager/1.0.0-preview2)

## .NET Compatibility
This release depends on .NET Standard 2.0 and runs on any .NET version which supports .NET Standard 2.0. The GA release will occur after the support end date for .NET Framework 4.6.1 ([April 26 2022](https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-framework)) so this release will support .NET Framework 4.6.2 and above and .NET Core 3.1 and above. It is built on top of ASP.NET Core, and has been tested and runs on all currently supported versions of ASP.NET Core up to 6.0.

## New Features
No new features in this release
 
## Bug Fixes
* Increased dependency version of Microsoft.IdentityModel.* packages to 6.17.0 (#598) - @birojnayak
* Fixed bug where a NullReferenceException could get thrown when using WebHttp if AddServiceModelWebServices hasn't been called (#601) - @tallalnparis4ev
* Fixed bug where wsdl and xsd URL query parameter values weren't parsed correctly (#607) - @roend83
* Created workaround for .NET runtime bugs when creating XSD schema for semantically equivalent data types (#605) - @mconnew 
* Removed unnecessary and broken static method SecurityTokenHandlerCollection.CreateDefaultSecurityTokenHandlerCollection() (#611) - @birojnayak

## Known Issues
Enum data types aren't correctly populated in the OpenAPI schema. The fix for this will be in the final release build.

**Full Changelog**: https://github.com/CoreWCF/CoreWCF/compare/v1.0.0-preview1...v1.0.0-preview2
 
## New Contributors
* @tallalnparis4ev made their first contribution in https://github.com/CoreWCF/CoreWCF/pull/601
* @roend83 made their first contribution in https://github.com/CoreWCF/CoreWCF/pull/607

## Feedback
Your feedback is important and appreciated. We've created a discussion item at https://github.com/CoreWCF/CoreWCF/discussions/ for your questions and comments.

<hr /><em>This discussion was created from the release <a href='https://github.com/CoreWCF/CoreWCF/releases/tag/v1.0.0-preview2'>v1.0.0-preview2</a>.</em>

**Comment by roend83** [comment]
I filed #615 with an issue I encountered after installing the Preview 2 package in a .NET 6 application.  I have a PR up (#616) that resolves the issue for me.

**Reply by mconnew** [reply]
Merged. This will be available in the GA release.

**Reply by roend83** [reply]
I have confirmed with that fix that I'm able to successfully generate a WSDL with no changes to my underlying service data model.
